File: src/main/kotlin/dev/jefy/connectpro/marketplace/applicaion/dtos/JobApplicationResponseForEmployer.java
package dev.jefy.connectpro.marketplace.applicaion.dtos;


import java.time.Instant;

import dev.jefy.connectpro.marketplace.domain.model.JobApplication;
import dev.jefy.connectpro.shared.domain.vo.JobApplicationStatus;
import dev.jefy.connectpro.user.applicaion.dtos.UserData;

/**
 * @author Jôph Yamba
 */
public record JobApplicationResponseForEmployer(
        UserData applicant,
        String motivation,
        Instant appliedAt,
        JobApplicationStatus status
) {
    public static JobApplicationResponseForEmployer fromDomain(
            JobApplication application,
            UserData applicant
    ) {
        return new JobApplicationResponseForEmployer(
                applicant,
                application.getMotivation(),
                application.getAppliedAt(),
                application.getStatus()
        );
    }
}

File: src/main/kotlin/dev/jefy/connectpro/marketplace/application/dtos/JobApplicationResponseForEmployer.kt
package dev.jefy.connectpro.marketplace.application.dtos

import dev.jefy.connectpro.marketplace.domain.JobApplication
import dev.jefy.connectpro.shared.domain.vo.JobApplicationStatus
import dev.jefy.connectpro.user.application.dtos.UserData
import java.time.Instant


/**
 * @author Jôph Yamba
 */

data class JobApplicationResponseForEmployer(
    val applicant: UserData,
    val motivation: String,
    val appliedAt: Instant,
    val status: JobApplicationStatus
)

fun JobApplication.toEmployerResponse(applicant: UserData) = JobApplicationResponseForEmployer(
    applicant = applicant,
    motivation = this.motivation,
    appliedAt = this.appliedAt,
    status = this.status
)
